lib/engine/route.rb

```diff
@@ -165,6 +165,22 @@ def check_connected!(token)
       # rubocop:enable Style/GuardClause, Style/IfUnlessModifier
     end
 
+    def check_terminals!
+      return unless paths.size > 2
```

Comment:
had to read this over a couple times, can we do

return if paths.size < 3

Reply:
Of course.